File: src/main/java/com/duxtinto/intellij/plugin/github/codereviews/di/impl/dagger/ApplicationContainer.java
package com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger;

import com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger.modules.ApplicationModule;
import com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger.modules.ProjectModule;
import dagger.Component;

import javax.inject.Singleton;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationContainer {
    ProjectContainer projectContainer(ProjectModule projectModule);
}

File: src/main/kotlin/com/duxtinto/intellij/plugin/github/codereviews/di/impl/dagger/ApplicationContainer.kt
package com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger

import com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger.modules.application.ApplicationModule
import com.duxtinto.intellij.plugin.github.codereviews.di.impl.dagger.modules.project.ProjectModule
import dagger.Component

import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class])
interface ApplicationContainer {
    fun projectContainer(projectModule: ProjectModule): ProjectContainer
}
